apps/server/src/modules/task-card/uc/task-card.uc.spec.ts

```diff
@@ -258,6 +258,49 @@ describe('TaskCardUc', () => {
 				await uc.create(user.id, failingTaskCardCreateParams);
 			}).rejects.toThrow(ValidationError);
 		});
+		it('should not throw if the date of dueDate and courseDate is the same, but they differ in time', async () => {
+			course = courseFactory.buildWithId({ untilDate: new Date(tomorrow.setHours(23, 58)) });
+			courseRepo.findById.mockResolvedValue(course);
+			taskCardCreateParams = {
+				title,
+				dueDate: new Date(tomorrow.setHours(23, 59)),
```

Comment:
to be honest this part made me think about the logic, as in my opinion it actually should throw if the dueDate is after the untilDate of the course, shouldnt it?
especially because you throw that error in the UC `Due date must be before course end date` which actually contradicts that it "should not throw" in this scenario where duedate is one minute after the course untilDate.
maybe its because its late already, but let me know about your thoughts on that

Reply:
clarified, missunderstanding from my side. it actually should work, that a dueDate can be in time before the course end date if its the on the same day.